src/relay.js

```diff
@@ -212,39 +218,22 @@ export function sequelizeConnection({name, nodeType, target, orderBy: orderByEnu
 
       if (args.after || args.before) {
         let cursor = fromCursor(args.after || args.before);
-        let orderValue = cursor.orderValue;
-
-        if (model.rawAttributes[orderAttribute].type instanceof model.sequelize.constructor.DATE) {
-          orderValue = new Date(orderValue);
-        }
-
-        let slicingWhere = {
-          $or: [
-            {
-              [orderAttribute]: {
-                [orderDirection === 'ASC' ? '$gt' : '$lt']: orderValue
-              }
-            },
-            {
-              [orderAttribute]: {
-                $eq: orderValue
-              },
-              [model.primaryKeyAttribute]: {
-                $gt: cursor.id
-              }
-            }
-          ]
-        };
+        let startIndex = Number(cursor.index);
 
         // TODO, do a proper merge that won't kill another $or
-        _.assign(options.where, slicingWhere);
+        if (startIndex > 0) options.offset = startIndex;
       }
-
       return before(options, args, root, context);
     },
     after: function (values, args, root, {source}) {
-      let edges = values.map((value) => {
-        return resolveEdge(value, args, source);
+      var cursor = null;
+
+      if (args.after || args.before) {
+        cursor = fromCursor(args.after || args.before);
+      }
+
+      let edges = values.map((value, idx) => {
+        return resolveEdge(value, idx + 1, cursor, args, source);
       });
```

Comment:
Would the index not be correct if we're already querying after a cursor? It would be 0 where it should be N + 0